app/controllers/api/manifests_controller.rb

```diff
@@ -1,8 +1,43 @@
 class Api::ManifestsController < ApplicationController
   def show
-    manifest_file = "public/manifest/#{ENV['SIMPLE_SERVER_ENV']}.json"
-    return head :not_found unless File.exists?(manifest_file)
-    render json: File.read(manifest_file),
-           status: :ok
+    if ENV['SIMPLE_SERVER_ENV'].in?(%w[development review])
+      render json: dynamic_manifest
+    else
+      manifest_file = "public/manifest/#{ENV['SIMPLE_SERVER_ENV']}.json"
+      return head :not_found unless File.exists?(manifest_file)
+
+      render json: File.read(manifest_file)
+    end
+  end
+
+  private
+
+  def dynamic_manifest
```

Comment:
This fragments the source of truth for the manifest. Can we instead just move the manifest to an ERB template (like `manifest.json.erb`) and always populate the `api_endpoint` from `ENV['SIMPLE_SERVER_HOST_PROTOCOL']` and `ENV['SIMPLE_SERVER_HOST']` and make them required config keys?

We can source the rest of the data in the manifest from `Rails.application.config.country`.

Reply:
We unfortunately can't because each of the "real" manifests (prod, demo) need to know of _every_ country's endpoints. I think these static files for them make sense. In the case of dev and review (and even SBX, I guess), all country endpoints point to the same box, but the endpoint of the box itself is context-dependent.

That being said, you're right about `erb` extraction. I'll do that.

Reply:
Extracted to a jbuilder in e19e10a

Reply:
Had no idea about `jbuilder`, pretty neat. I'm still a little concerned about the split, but I understand the limitation.

Do you think we can write a simple regression test that ensures that review-env keys in the manifest match the real-env keys?

Reply:
@kitallis Yeah it kinda sucks that they're conceptually different now.

By regression test, do you mean like this one here? https://github.com/simpledotorg/simple-server/pull/986/files#diff-858391f632a627cb72d50e9884e1d8e1R41

Or did you have something else in mind?